lib/manager/poetry/artifacts.ts

```diff
@@ -107,8 +116,10 @@ function getSourceCredentialVars(
   const envVars: Record<string, string> = {};
 
   for (const source of poetrySources) {
-    const matchingHostRule = find({ url: source.url });
-    const formattedSourceName = source.name.toUpperCase();
+    const matchingHostRule = getMatchingHostRule(source);
+    const formattedSourceName = source.name
```

Comment:
Do we validate the source, so we can't get null or undefined references here?

Reply:
Yes, at this point we know that the source has a name and an URL: https://github.com/renovatebot/renovate/blob/main/lib/manager/poetry/artifacts.ts#L95